p2p/pex/reactor.go

```diff
@@ -389,6 +401,8 @@ func (r *ReactorV2) waitUntilNextRequest() <-chan time.Time {
 // peer into the requestsSent bucket and calculates when the next request
 // time should be
 func (r *ReactorV2) sendRequestForPeers() {
+	r.mtx.Lock()
```

Comment:
Was this move to handle `peer := r.availablePeers.Front()`?

Reply:
yeah we need a mutex over the entire function because the peer could be removed by the peerUpdates go routine before we send the request